python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -601,11 +731,12 @@ def get_asset_provenance(self, asset_key: AssetKey) -> Optional[DataProvenance]:
                 materialization of the asset. Returns `None` if the asset was never materialized or
                 the materialization record is too old to contain provenance information.
         """
-        record = self.instance.get_latest_data_version_record(asset_key)
-
-        return (
-            None if record is None else extract_data_provenance_from_entry(record.event_log_entry)
-        )
+        return super().get_asset_provenance(asset_key)
 
 
-AssetExecutionContext: TypeAlias = OpExecutionContext
+def build_execution_context(
+    step_context: StepExecutionContext,
+) -> Union[OpExecutionContext, AssetExecutionContext]:
+    if step_context.step_materializes_assets:
```

Comment:
name predates you, but I actually think this name is quite deceiving, as ops can yield `AssetMaterialization` objects.

Reply:
ya i had a similar reaction - what do you think would be better? this is internal so we can change it